test/internet/test-dns-ipv4.js

```diff
@@ -0,0 +1,203 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert'),
+    dns = require('dns'),
+    net = require('net'),
+    isIP = net.isIP,
+    isIPv4 = net.isIPv4;
+var util = require('util');
+
+var expected = 0,
+    completed = 0,
+    running = false,
+    queue = [];
+
+function TEST(f) {
```

Comment:
As far as `TEST()` goes, I was torn between "move this into its own module" vs. "extract each test into its own file and get rid of `TEST()` entirely". So I left it alone for now.